pal/src/map/common.cpp

```diff
@@ -49,7 +49,7 @@ INT W32toUnixAccessControl( IN DWORD flProtect )
         MemAccessControl = PROT_EXEC | PROT_READ | PROT_WRITE;
         break;
     case PAGE_EXECUTE :
-        MemAccessControl = PROT_EXEC;
+        MemAccessControl = PROT_EXEC | PROT_READ; // WinAPI PAGE_EXECUTE also implies readable
```

Comment:
> Since on Windows : https://msdn.microsoft.com/en-us/library/windows/desktop/aa366786(v=vs.85).aspx, PAGE_EXECUTE on Windows implies readable, only writes will generate AV. 

Well, the problem is that I don't see any promise regarding to `READ` access on that document. Isn't `PAGE_EXECUTE_READ` supposed to be used for that?

Even more, memory dump tools show different marks for PAGE_EXECUTE_READ and PAGE_EXECUTE on a particular region. IMHO, PAL matches to Windows API. The problem here is that we read on xplat while we don't on Windows. This PR showed us that even if we would read that area, we could do it without an issue.

I don't have the environment to test it; we have very few `PAGE_EXECUTE` usage in the code, can we test which one is causing this problem and replace only that/those to `PAGE_EXECUTE_READ` instead?